src/compiler/checker.ts

```diff
@@ -8765,7 +8765,7 @@ export function createTypeChecker(host: TypeCheckerHost): TypeChecker {
                                     /*modifiers*/ undefined,
                                     getEffectiveDotDotDotForParameter(p),
                                     setTextRange(context, factory.createIdentifier(getNameForJSDocFunctionParameter(p, i)), p),
-                                    p.questionToken,
+                                    p.questionToken ? factory.createToken(SyntaxKind.QuestionToken) : undefined,

```

Comment:
Some other code paths in this function do a similar "cloning" of the `.questionToken` so I assume that it's the right thing to do here and in other places that I touched

Reply:
Those 2 call sites related to `.questionToken` are also related to JSDoc. I wonder if this cloning even matters in this context.